src/main/java/forum/dao/ThreadDAOImpl.java

```diff
@@ -0,0 +1,40 @@
+package forum.dao;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import forum.entity.Thread;
+
+@Repository
+public class ThreadDAOImpl implements ThreadDAO {
+	
+	@Autowired
+	private SessionFactory sessionFactory;
+
+	@Override
+	public void saveThread(Thread thread) {
+		
+		Session currentSession = sessionFactory.getCurrentSession();
+		
+		currentSession.saveOrUpdate(thread);
+		
+	}
+
+	@Override
+	public List<Thread> getThreads() {
+		
+		Session currentSession = sessionFactory.getCurrentSession();
+		
+		Query<Thread> theQuery = currentSession.createQuery("from Thread", Thread.class);
```

Comment:
Can you try to use Criteria API to make queries since it is more robust and you don't have to do messy string concat in more complicated queries. More info here: https://www.baeldung.com/hibernate-select-all
Also check this: https://thoughts-on-java.org/hibernate-best-practices/

Reply:
I will give it a try!